specification/communication/data-plane/JobRouter/preview/2021-10-20-preview2/communicationservicejobrouter.json

```diff
@@ -0,0 +1,3635 @@
+          },
+          {
+            "in": "header",
+            "name": "repeatability-first-sent",
+            "type": "string"
+          },
+          {
+            "in": "body",
+            "name": "classificationPolicy",
+            "description": "Model of classification policy to create",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ClassificationPolicy"
+            }
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/ClassificationPolicy"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Creates a Classification policy": {
+            "$ref": "./examples/ClassificationPolicies_CreateClassificationPolicy.json"
+          }
+        }
+      },
+      "get": {
+        "tags": [
+          "ClassificationPolicies"
+        ],
+        "summary": "Retrieves existing classification policies",
+        "operationId": "JobRouter_ListClassificationPolicies",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "query",
+            "name": "maxpagesize",
+            "description": "Maximum page size",
+            "type": "integer",
+            "format": "int32",
+            "default": 20
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/ClassificationPolicyCollection"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get Classification policies with pagination (max page size)": {
+            "$ref": "./examples/ClassificationPolicies_ListClassificationPoliciesWithPageSize.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink",
+          "itemName": "value"
+        }
+      }
+    },
+    "/routing/classificationPolicies/{id}": {
+      "patch": {
+        "tags": [
+          "ClassificationPolicies"
+        ],
+        "summary": "Updates a classification policy.",
+        "operationId": "JobRouter_PatchClassificationPolicyV2",
+        "consumes": [
+          "application/merge-patch+json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "path",
+            "name": "id",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "patch",
+            "description": "JSON Document contained the partial update for the classification policy",
+            "schema": {
+              "$ref": "#/definitions/ClassificationPolicy"
+            }
```

Comment:
This is one of many operations with a body parameter that is not defined as `required: true`.  This is typically an oversight and you should consider adding `required: true` unless there really is a use case for omitting a body from the request.

Reply:
Tracked using [issue](https://skype.visualstudio.com/SPOOL/_workitems/edit/2788064)

Reply:
JSON MergePatch accepts a an empty patch object `{}` (but not `null`) and does no operation which is behavior our service matches. However I imagine nobody would want to intentionally make this api call as it doesn't update anything and would effectively be a get operation. To avoid this, SDK's should probably want to avoid an optional patch argument, but this can be enforced by the helper methods in the SDK's.

So, should the swagger be reflective of the SDK's, and have `required: true` for `patch`, or reflective of the service which patch arguments can be empty.

Reply:
@rcoleobrien An empty patch object _is_ a request body, so that's not the concern here.  The question is: what happens when no body is sent at all, e.g. `content-length: 0`. Does the service allow this or reject it with an error response?

Reply:
The service rejects with an error when no body is sent